### PR TITLE
Treat all warnings as errors (non wasm)

### PR DESCRIPTION
**Pull request by mcbarton**
The PR will change it so when building xeus-cpp it treats all warnings as errors by default. In the ci I turn this off for wasm builds.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -47,6 +47,8 @@ OPTION(XEUS_CPP_BUILD_EXECUTABLE "Build the xcpp executable" ON)
 OPTION(XEUS_CPP_USE_SHARED_XEUS "Link xcpp  with the xeus shared library (instead of the static library)" ON)
 OPTION(XEUS_CPP_USE_SHARED_XEUS_CPP "Link xcpp  with the xeus shared library (instead of the static library)" ON)
 
+option(CMAKE_COMPILE_WARNING_AS_ERROR "Compile warnings as errors" ON)
```
Comment (JohanMabille) [inline CMakeLists.txt:50]: - No need to create an option here, `CMAKE_COMPILE_WARNING_AS_ERROR` already exists as a [cmake variable](https://cmake.org/cmake/help/latest/variable/CMAKE_COMPILE_WARNING_AS_ERROR.html) and can be passed directly in the command line.
- Although it's a good thing to have it enabled by default on the CI, it should be off by default for other users
- The required minimal version of cmake should be updated to be compatible with the use of this variable.

File: src/xinterpreter.cpp [hunk 2]
```diff
@@ -138,6 +138,9 @@ __get_cxx_version ()
     {
         nl::json kernel_res;
 
+
+        auto input_guard = input_redirection(allow_stdin);
```
Comment (JohanMabille) [inline src/xinterpreter.cpp:142]: Why adding this line in this PR? This seems unrelated.

**Review by JohanMabille — CHANGES_REQUESTED** [review]

File: src/xinterpreter.cpp (hunk 2 above)
Comment (mcbarton) [reply src/xinterpreter.cpp:142]: This is to fix a warning which comes when building xeus-cpp. Without it allow_stdin is an unused variable. I coped how its used in xeus-python.

File: CMakeLists.txt (hunk 1 above)
Comment (mcbarton) [reply CMakeLists.txt]: This comment has now been addressed.

**Comment by mcbarton** [comment]
@JohanMabille This PR is ready for another review. I have addressed your comments.

**Comment by JohanMabille** [comment]
Thanks!

**Review by JohanMabille — APPROVED** [review]
